### PR TITLE
Origin/feature/fave 17276 progress

### PR DESCRIPTION
**Pull request by AladdinNullptr**
# Description

Adds progress component

## Type of change

Please delete options that are not relevant.
- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings

### CODE REVIEW

File: src/Progress/index.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+export type ProgressProps = AntProgressProps
+
+// AntProgress does NOT support passing icons
+const Progress: React.FC<ProgressProps> = ({ size, strokeWidth, ...restProps }) => {
+  const defaultStrokeWidth = strokeWidth || (size && size === 'small' ? 4 : 6)
+
+  return <AntProgress strokeWidth={defaultStrokeWidth} size={size} {...restProps} />
+}
+
+Progress.defaultProps = {}
```
Comment (alex-safian) [inline src/Progress/index.tsx:15]: Remove this line this if the component doesn't have any default props.

File: src/Progress/style.css [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+          content: var(--ic-check-16);
+        }
+      }
+    }
+
+    &.ant-progress-line {
+      & .ant-progress-bg {
+        @apply bg-green-500;
+      }
+      & .ant-progress-text {
```
Comment (alex-safian) [inline src/Progress/style.css:68]: Add a line break here.

File: src/Progress/style.css [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+          content: var(--ic-x-16);
+        }
+      }
+    }
+
+    &.ant-progress-line {
+      & .ant-progress-bg {
+        @apply bg-red-600;
+      }
+      & .ant-progress-text {
```
Comment (alex-safian) [inline src/Progress/style.css:89]: Add a line break here.

File: src/Progress/style.css [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+        }
+      }
+    }
+  }
+}
+
+.ant-progress {
+  &.ant-progress-status-success.ant-progress-default {
+    &.ant-progress-circle {
+      & .ant-progress-text {
```
Comment (alex-safian) [inline src/Progress/style.css:37]: Is it possible to reduce the number of selectors? If yes, please do the same thing for other selector too.

**Comment by alex-safian** [comment]
Please fix the branch name. we shouldn't add origin to our branch names.

File: src/Progress/style.css (hunk 4 above)
Comment (AladdinNullptr) [reply src/Progress/style.css:37]: The original selector is specific. So no. I can change it to one line but it would be difficult to modify

**Pull request closed without merging**